src/ArrayComparator.php

```diff
@@ -124,4 +125,33 @@ private function indent(string $lines): string
     {
         return trim(str_replace("\n", "\n    ", $lines));
     }
+
+    private function canonicalize(array &$array): void
+    {
+        if ($this->isIndexedArray($array)) {
+            sort($array);
+        } else {
+            ksort($array);
+        }
+
+        foreach ($array as &$element) {
+            if (is_array($element)) {
+                $this->canonicalize($element);
+            }
+        }
+    }
+
+    private function isIndexedArray(array $array): bool
```

Comment:
You should be able to replace this method with [`array_is_list()`](https://www.php.net/manual/en/function.array-is-list.php).

Reply:
Wow. Thanks. Fixed.